Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -82,27 +143,46 @@
 }
 
 void
-sentry_discardProfilerForTracer(SentryId *internalTraceId)
+sentry_discardProfilerHybrid(SentryId *internalTraceId, SentryHub *hub)
+{
+}
+
+void
+sentry_discardProfiler(SentryId *internalTraceId, SentryHub *hub, BOOL traceSampled)
 {
     std::lock_guard<std::mutex> l(_gStateLock);
 
-    SENTRY_CASSERT(_gTracersToProfilers != nil && _gProfilersToTracers != nil,
-        @"Structures should have already been initialized by the time they are being queried");
+    if ([hub.getClient.options isContinuousProfilingEnabled]) {
+        if (hub.getClient.options.profiling.lifecycle != SentryProfileLifecycleTrace) {
+            return;
+        }
+        if (!traceSampled) {
+            return;
+        }
+        _unsafe_cleanUpContinuousTraceProfiler();
+    } else {
+        if (internalTraceId == nil) {
+            return;
```

Comment:
`l`: Does this maybe deserve an extra log message, as this seems like an edge case? Maybe I'm also mistaken.

Reply:
yeah i'm always good with more logging!